packages/blocks/src/blocks/CoreImage.tsx

```diff
@@ -0,0 +1,104 @@
+import { gql } from '@apollo/client';
+import React from 'react';
+import Image from 'next/image.js';
+import { useBlocksTheme } from '../components/WordPressBlocksProvider.js';
+import { ContentBlock } from '../components/WordPressBlocksViewer.js';
+import { getStyles } from '../utils/index.js';
+
+export type CoreImageFragmentProps = ContentBlock & {
+  attributes: {
+    align?: string;
+    alt?: string;
+    anchor?: string;
+    borderColor?: string;
+    caption?: string;
+    className?: string;
+    width?: string;
+    url?: string;
+    title?: string;
+    style?: string;
+    src?: string;
+    sizeSlug?: string;
+    rel?: string;
+    lock?: string;
+    linkTarget?: string;
+    linkDestination?: string;
+    linkClass?: string;
+    id?: string;
+    href?: string;
+    height?: string;
+    cssClassName?: string;
+  };
+};
+
+export function CoreImage(props: CoreImageFragmentProps) {
+  const theme = useBlocksTheme();
+  const style = getStyles(theme, { ...props });
+  const { attributes } = props;
+
+  if (attributes.width && attributes.height) {
+    return (
+      <figure>
```

Comment:
the `cssClassName` should be in the `figure` and not in the `Image`:

```suggestion
      <figure className={attributes?.cssClassName}>
```